tools/dds/dds-server/dds-device-broadcaster.cpp

```diff
@@ -53,100 +57,85 @@ dds_server::dds_server()
     , _ctx( "{"
             "\"dds-discovery\" : false"
             "}" )
+    , init_done( false )
+
 {
 }
-bool dds_server::init( DomainId_t domain_id )
+bool dds_device_broadcaster::init()
 {
-    auto participant_created_ok = create_dds_participant( domain_id );
-    auto publisher_created_ok = create_dds_publisher();
-    return participant_created_ok && publisher_created_ok;
+    if( !_participant )
+    {
+        return false;
+    }
+
+    init_done = create_dds_publisher();
+    return init_done;
 }
 
-void dds_server::run()
+void dds_device_broadcaster::run()
 {
+    if( !init_done )
+    {
+        std::cout << "RS DDS Server was not initialize, please call 'init()' first" << std::endl;
```

Comment:
Looks like this should be an exception, no?

Reply:
Changed as discussed

Reply:
And the exception?

Reply:
`run()` return a bool.
we print an error message and return false.
isn't it enough? init content is now inside the run.
so we dont have this case anymore (run called before init)

Reply:
We can do a dedicated PR after this one, lets seperate